language/move-core/types/src/state.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::cell::RefCell;
+
+#[derive(Clone, Copy, Debug)]
+pub enum VMState {
+    DESERIALIZER,
+    VERIFIER,
+    RUNTIME,
+    OTHER,
```

Comment:
I don't know about `OTHER` I think a more meaningful state that indicates we are not inside a catch unwind would be great. Something like `YO_NOT_IN_A_CATCH_UNWIND_YOU_YO` the point I was trying to make is the following:
a thread enters the VM, hits deserializer and verifier and there is no problem. The thread local is now left with `OTHER` later that thread panics and a global panic handler picks up and checks the thread local, if it finds `OTHER` that means there is no catch unwind and the global handler can do as it wishes to do.
So `OTHER` really means not in a catch unwind but it hardly reads that way.
BTW I love `YO_NOT_IN_A_CATCH_UNWIND_YOU_YO` and if we put it there I will accept immediately